lesson18/main.js

```diff
@@ -77,9 +77,7 @@ const createNextBtn = () => {
   nextBtn.addEventListener(
     "click",
     () => {
-      switchImg("nextElementSibling");
-      switchDisableForBtn();
-      getCurrentPageNum();
+      clickBtnEvents("nextElementSibling");
```

Comment:
createNextBtnがcreate以外もやっているので名前を変えるか処理を分けた方がいい気がしました

Reply:
[ここ](https://codesandbox.io/embed/lesson18-create-dot-pagination-forked-iukl1?fontsize=14&hidenavigation=1&theme=dark)にここのコメントに対する修正の一端のゴールを示します。わからない場合見てみてください

Reply:
You are right, it doesn't match the name. Thank you also for the reference code! :)
I've tried by my self first and it became like this ↓
```
const createArrowBtnForSlideshow = () => {
  createNextBtn();
  addEventToBtn("nextElementSibling", "js-next-btn");
  createBackBtn();
  addEventToBtn("previousElementSibling", "js-back-btn");
};

// const createNextBtnWithEvent = () => {
//   createNextBtn();
//   addEventToBtn("nextElementSibling", "js-next-btn");
// };

// const createBackBtnWithEvent = () => {
//   createBackBtn();
//   addEventToBtn("previousElementSibling", "js-back-btn");
// };

const createNextBtn = () => {
  const nextBtn = createElementWithClassName("button", "btn");
  nextBtn.id = "js-next-btn";
  nextBtn.textContent = ">";
  ul.insertAdjacentElement("afterend", nextBtn);
};

const createBackBtn = () => {
  const backBtn = createElementWithClassName("button", "btn");
  backBtn.id = "js-back-btn";
  backBtn.textContent = "<";
  backBtn.disabled = true;
  ul.insertAdjacentElement("beforebegin", backBtn);
};

const addEventToBtn = (direction, id) => {
  const btn = document.getElementById(id);
  btn.addEventListener(
    "click",
    () => {
      clickBtnEvents(direction);
    },
    false
  );
};

const clickBtnEvents = (switchDirection) => {
  switchImgForBtn(switchDirection);
  changeIndicator(switchDirection);
  switchDisableForBtn();
  getCurrentPageNum();
};
```
At first I implemented `createNextBtnWithEvent` etc and called them in `createArrowBtnForSlideshow`.But I thought it's a little too detailed and confusing while I was trying, so I changed.
What do you think this code?

After trying refactoring I checked your code, and I learned better way! I can make event part in `createArrowBtnForSlideshow` and it looks organized without confusing.

Reply:
```js
clickBtnEvents(direction),
.
.
.
const clickBtnEvents = (switchDirection) => () => { switchImgForBtn(switchDirection); changeIndicator(switchDirection); switchDisableForBtn(); getCurrentPageNum(); };
```

Reply:
Thank you. I agree, create doesn't include output.
I've implemented other parts like `loader`, `image`, `pagination`, `indicator`, but I've also used `create` for their function name. I'm thinking I'm going to change their name `output` instead of `create`, what do you think? Is this name in common?

And I'm sorry, but I don't understand the second one what you taught me. Would you explain it in more detail? Thank you.

Reply:
Thank you for the reference url. This time I chose `render`, I think it sounds more natural.
I've changed some functions name which create an element something and not return anything.
[2aa189d](https://github.com/yuka830/js-lessons/pull/23/commits/2aa189d4c9677a9b7309324425c9a9971fe9efd6)